runtime/complex.go

```diff
@@ -130,3 +131,40 @@ func complexCoerce(o *Object) (complex128, bool) {
 	}
 	return complex(floatO, 0.0), true
 }
+
+func complexArithmeticOp(f *Frame, method string, v, w *Object, fun func(v, w *Object) complex128) (*Object, *BaseException) {
+	return NewComplex(fun(v, w)).ToObject(), nil
+}
+
+func complexAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
+	if w.isInstance(ComplexType) {
+		return complexArithmeticOp(f, "__add__", v, w, complexAddComplex)
+	}
+
+	if w.isInstance(IntType) {
```

Comment:
I think if w is not complex then it's cleaner to use floatCoerce(w). That will also support long properly.